app/models/info_request.rb

```diff
@@ -1201,6 +1201,20 @@ def indexed_by_search?
     true
   end
 
+  def self.set_reject_mail_at_mta(options)
```

Comment:
This doesn't return the array if `dryrun` is `false`. I'm not actually sure why, because `update_all` didn't seem to exhibit this behaviour while having a play around in the console.

In any case, I think yielding the IDs of records to be updated would be better:

```diff
diff --git a/app/models/info_request.rb b/app/models/info_request.rb
index a673496..2d1bc6e 100644
--- a/app/models/info_request.rb
+++ b/app/models/info_request.rb
@@ -1210,10 +1210,14 @@ class InfoRequest < ActiveRecord::Base
                                 AND reject_mail_at_mta = 'f'
                                 AND url_title <> 'holding_pen'",
                                 options[:rejection_threshold]])
+
+    yield query.pluck(:id) if block_given?
+
     if !options[:dryrun]
       query.update_all("reject_mail_at_mta = 't'")
+    else
+      0
     end
-    query.to_a
   end

   # This is called from cron regularly.
```

- The return value is always the number of records updated
- You avoid loading all the records in to memory

```ruby
updated_count = InfoRequest.set_reject_mail_at_mta(options) do |ids|
  puts "Info Request\tRejected incoming count\tLast updated"
  ids.each do |id|
    info_request = InfoRequest.find(id)
    puts "#{info_request.id}\t#{info_request.rejected_incoming_count}\t#{info_request.updated_at}"
  end
end
puts "Updated #{updated_count} info requests"
```

Some other miner comments:

We're a bit beyond supporting MySQL now, but a hash argument would read better here:

```ruby
query.update_all(:reject_mail_at_mta => true)
```

Could also use `unless options[:dryrun]` instead of `if !options[:dryrun]`.

Super minor, but I'd drop the `set_`.